lib/GraphReader.cc

```diff
@@ -163,54 +165,80 @@ void GraphReader::read_partition(BipartiteGraph::ContainerType& vmap) {
     match(TOK_AT);
     match(TOK_END);
 }
-
 /// @PreferenceLists (A|B)
 /// preference lists
 /// @End
 /// preference lists for a vertex are given in this format
-/// v: a, b, c ;
+/// v: a, (b, c), d ;
 /// we read the preference lists for vertices in partition A
 /// and lookup the other side in partition B
 void GraphReader::read_preference_lists(BipartiteGraph::ContainerType& A, BipartiteGraph::ContainerType& B) {
     // read the lists
     while (curtok_ != TOK_AT) {
         // read the vertex for which the preference list is given
         std::string a = lexer_->get_lexeme();
-        match(TOK_STRING);
+        match(TOK_STRING); //consume the string because you have already stored it in a 
```

Comment:
This sentence can be reworded as "Consume string token because copy is saved in a."